packages/block-library/src/query/index.php

```diff
@@ -123,3 +123,87 @@ function register_block_core_query() {
 	);
 }
 add_action( 'init', 'register_block_core_query' );
+
+/**
+ * Traverse the tree of blocks looking for any plugin block (i.e., a block from
+ * an installed plugin) inside a Query block with the enhanced pagination
+ * enabled. If at least one is found, the enhanced pagination is effectively
+ * disabled to prevent any potential incompatibilities.
+ *
+ * @since 6.4.0
+ *
+ * @param array $parsed_block The block being rendered.
+ *
+ * @return string Returns the parsed block, unmodified.
+ */
+function block_core_query_disable_enhanced_pagination( $parsed_block ) {
+	static $enhanced_query_stack   = array();
+	static $dirty_enhanced_queries = array();
+	static $render_query_callback  = null;
+
+	$block_name = $parsed_block['blockName'];
+
+	if (
+		'core/query' === $block_name &&
+		isset( $parsed_block['attrs']['enhancedPagination'] ) &&
```

Comment:
I don't think it is defined when it's not set. So for safety, and because this is meant to be merged in the last RC, I'd keep the `isset()`:

<img width="203" alt="Screenshot 2023-11-01 at 09 56 13" src="https://github.com/WordPress/gutenberg/assets/3305402/c84a9103-e6be-457b-8594-90878829b127">